vg/vgimg/vgimg.go

```diff
@@ -40,46 +44,73 @@ type Canvas struct {
 // New returns a new image canvas with
 // the size specified  rounded up to the
 // nearest pixel.
-func New(width, height vg.Length) *Canvas {
-	w := width / vg.Inch * dpi
-	h := height / vg.Inch * dpi
+func New(width, height vg.Length, options ...func(*Canvas)) *Canvas {
+	c := new(Canvas)
+	c.dpi = defaultDPI
+	for _, option := range options {
+		option(c)
+	}
+	w := width / vg.Inch * vg.Length(c.dpi)
+	h := height / vg.Inch * vg.Length(c.dpi)
 	img := image.NewRGBA(image.Rect(0, 0, int(w+0.5), int(h+0.5)))
 
-	return NewImage(img)
+	return NewImage(img, options...)
```

Comment:
I don't like how this evaluates the options multiple times. I realize that it's not causing a problem here, but I still think that it would be better not to get into this habit. Instead, we should factor out code that's currently shared by delegating to the next `New` call into separate functions that are called by each `New`:
```
func makeGraphicsContext(img draw.Image, dpi int) draw2d.GraphicsContext {
  gc := draw2d.NewGraphicsContext(img)
  gc.SetDPI(dpi)
  gc.Scale(1, -1)
  gc.Translate(0, -h)
  return gc
}

func initCanvas(c *Canvas, img draw.Image, gc draw2d.GraphicsContext) {
  w := float64(img.Bounds().Max.X - img.Bounds().Min.X)
  h := float64(img.Bounds().Max.Y - img.Bounds().Min.Y)	
  draw.Draw(img, img.Bounds(), image.White, image.ZP, draw.Src)
  c.w = vg.Length(w/float64(c.dpi)) * vg.Inch
  ...
}
```